File: knightsTour/Backtracking.cs
using knightsTour.Model;
using System;
using System.Collections.Generic;
using System.Text;

namespace knightsTour
{
    public class Backtracking : KnightsTourAlgorithm
    {
        public Backtracking(Chessboard chessboard) : base(chessboard)
        {
        }

        public bool solveKT()
        {

            if (solveKTRecursion(Chessboard.Board, 1, Chessboard.KnightX, Chessboard.KnightY))
            {
                printSolution(Chessboard, null);
            }
            else
            {
                Console.WriteLine("Solution for the given problem does not exist");
            }

            return true;
        }

        private bool solveKTRecursion(int[,] board, int iteration, int knightX, int knightY)
        {
            //printSolution(null, board);
            if (isFinished(iteration, board))
            {
                return true;
            }

            HashSet<Move> viableMoves = MovesService.GenerateLegalMoves(null, board, knightX, knightY);

            foreach (Move move in viableMoves)
            {
                int nextX = knightX + move.X;
                int nextY = knightY + move.Y;

                board[nextY, nextX] = iteration + 1;

                if (solveKTRecursion(board, iteration + 1, nextX, nextY))
                {
                    return true;
                }
                else
                {
                    board[nextY, nextX] = 0;
                }

            }

            return false;
        }
    }
}

File: Deprecated_Code_In_Java/src/main/java/com/knightstour/Backtracking.java
package com.knightstour;

import com.knightstour.model.Chessboard;
import com.knightstour.model.Moves;

import java.util.List;

public class Backtracking extends KnightsTourAlgorithmImpl {

    public Backtracking(Chessboard chessboard) {
        super(chessboard, Backtracking.class.toString());
    }

    public boolean solveKT() {

        if (solveKTRecursion(getChessboard().getBoard(), 1, getChessboard().getKnightX(), getChessboard().getKnightY())) {
            printSolution(getChessboard());
        } else {
            System.out.println("Solution for the given problem does not exist");
        }

        return true;
    }

    private boolean solveKTRecursion(int[][] board, int iteration, int knightX, int knightY) {

        //log.info(Integer.toString(iteration));

        if (isFinished(iteration, board)) {
            return true;
        }

        List<Moves> viableMoves = generateLegalMoves(board, knightX, knightY);

        for (Moves move : viableMoves) {
            int nextX = knightX + move.getX();
            int nextY = knightY + move.getY();

            board[nextX][nextY] = iteration + 1;

            if (solveKTRecursion(board, iteration + 1, nextX, nextY)) {
                return true;
            } else {
                board[nextX][nextY] = 0;
            }

        }

        return false;
    }

}
